src/workerd/jsg/util.c++

```diff
@@ -570,29 +571,60 @@ kj::Array<kj::byte> asBytes(v8::Local<v8::ArrayBufferView> arrayBufferView) {
 }
 
 void recursivelyFreeze(v8::Local<v8::Context> context, v8::Local<v8::Value> value) {
-  if (value->IsArray()) {
-    // Optimize array freezing (Array is a subclass of Object, but we can iterate it faster).
-    v8::HandleScope scope(v8::Isolate::GetCurrent());
-    auto arr = value.As<v8::Array>();
-
-    for (auto i: kj::zeroTo(arr->Length())) {
-      recursivelyFreeze(context, check(arr->Get(context, i)));
-    }
+  auto isolate = v8::Isolate::GetCurrent();
+  v8::HandleScope outerScope(isolate);
+  kj::Vector<v8::Local<v8::Value>> queue;
+  queue.reserve(128);
+  queue.add(value);
+
+  while (!queue.empty()) {
+    auto item = queue.back();
+    queue.removeLast();
```

Comment:
I think this might be a problem with the original too, but how is this going to handle circular references?

Reply:
It won't handle circular references. The util.h has a comment saying that it's ok to ignore circular references. This method is called on a data that is parsed and stringified, hence eliminating circular references.

Reply:
that's what I figured. Let's add a comment in here mentioning that.

Reply:
The input could have circular references. Such input is _invalid_, because it will later attempt to be serialized to JSON. So it's OK if we throw an exception in that case, but we can't completely ignore it.

Reply:
Well, currently our only use of this is an object that comes straight out of a JSON.parse, and the current impl doesn't handle circular references either. Still, would be better to handle if we want this to be generic.

Reply:
Oops, I guess I misremembered where this is used, sorry.